src/connector/src/connector_common/postgres.rs

```diff
@@ -0,0 +1,450 @@
+                SslMode::Disabled => PgSslMode::Disable,
+                SslMode::Preferred => PgSslMode::Prefer,
+                SslMode::Required => PgSslMode::Require,
+                SslMode::VerifyCa => PgSslMode::VerifyCa,
+                SslMode::VerifyFull => PgSslMode::VerifyFull,
+            });
+
+        if *ssl_mode == SslMode::VerifyCa || *ssl_mode == SslMode::VerifyFull {
+            if let Some(ref root_cert) = ssl_root_cert {
+                options = options.ssl_root_cert(root_cert.as_str());
+            }
+        }
+
+        let connection = PgPool::connect_with(options).await?;
+        let schema_discovery = SchemaDiscovery::new(connection, schema);
+        // fetch column schema and primary key
+        let empty_map = HashMap::new();
+        let table_schema = schema_discovery
+            .discover_table(
+                TableInfo {
+                    name: table.to_string(),
+                    of_type: None,
+                },
+                &empty_map,
+            )
+            .await?;
+
+        let mut column_name_to_pg_type = HashMap::new();
+        let mut column_descs = vec![];
+        for col in &table_schema.columns {
+            let data_type = type_to_rw_type(&col.col_type)?;
+            let column_desc = if let Some(ref default_expr) = col.default {
+                // parse the value of "column_default" field in information_schema.columns,
+                // non number data type will be stored as "'value'::type"
+                let val_text = default_expr
+                    .0
+                    .split("::")
+                    .map(|s| s.trim_matches('\''))
+                    .next()
+                    .expect("default value expression");
+
+                match ScalarImpl::from_text(val_text, &data_type) {
+                    Ok(scalar) => ColumnDesc::named_with_default_value(
+                        col.name.clone(),
+                        ColumnId::placeholder(),
+                        data_type.clone(),
+                        Some(scalar),
+                    ),
+                    Err(err) => {
+                        tracing::warn!(error=%err.as_report(), "failed to parse postgres default value expression, only constant is supported");
+                        ColumnDesc::named(col.name.clone(), ColumnId::placeholder(), data_type)
+                    }
+                }
+            } else {
+                ColumnDesc::named(col.name.clone(), ColumnId::placeholder(), data_type)
+            };
+            {
+                let pg_type = Self::discovered_type_to_pg_type(&col.col_type)?;
+                column_name_to_pg_type.insert(col.name.clone(), pg_type);
+            }
+            column_descs.push(column_desc);
+        }
+
+        if !is_append_only && table_schema.primary_key_constraints.is_empty() {
+            return Err(anyhow!(
+                "Postgres table should define the primary key for non-append-only tables"
```

Comment:
I think this line leads to the error @xxchan.

🤔 maybe sea-orm could not discover the PK for some reason..

Reply:
But actually this check already exists previously. Not sure why this bug newly occurs.

Reply:
https://github.com/risingwavelabs/risingwave/pull/19328/files#r1970903433